cmds/core/find/find.go

```diff
@@ -17,84 +17,97 @@ import (
 	"context"
 	"flag"
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"strings"
 
+	"github.com/u-root/u-root/pkg/uroot/util"
+
 	"github.com/u-root/u-root/pkg/find"
 )
 
 const cmd = "find [opts] starting-at-path"
 
-var (
-	perm     = flag.Int("mode", -1, "Permissions")
-	fileType = flag.String("type", "", "File type")
-	name     = flag.String("name", "", "glob for name")
-	long     = flag.Bool("l", false, "long listing")
-	debug    = flag.Bool("d", false, "Enable debugging in the find package")
-)
-
-func init() {
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
-		os.Exit(1)
-	}
+type flags struct {
+	perm     int
+	filetype string
+	name     string
+	long     bool
+	debug    bool
 }
 
-func main() {
-	fileTypes := map[string]os.FileMode{
+var (
+	fargs     = flags{}
+	fileTypes = map[string]os.FileMode{
 		"f":         0,
 		"file":      0,
 		"d":         os.ModeDir,
 		"directory": os.ModeDir,
 	}
+)
 
-	flag.Parse()
-	a := flag.Args()
-	if len(a) != 1 {
+func init() {
+	flag.IntVar(&fargs.perm, "mode", -1, "Permissions")
+	flag.StringVar(&fargs.filetype, "type", "", "File type")
+	flag.StringVar(&fargs.name, "name", "", "glob for name")
+	flag.BoolVar(&fargs.long, "l", false, "long listing")
+	flag.BoolVar(&fargs.debug, "d", false, "Enable debugging in the find package")
+}
+
+func runFind(out io.Writer, errOut io.Writer, fargs flags, arg []string) error {
+	if len(arg) != 1 {
```

Comment:
well guess what :-)
0 args is legal. Should we do that?

Reply:
I'm not sure, but you do realize the CI failed?